core-bundle/config/listener.yaml

```diff
@@ -253,6 +265,15 @@ services:
         tags:
             - kernel.event_listener
 
+    contao.listener.finalize_response_context:
+        class: Contao\CoreBundle\EventListener\FinalizeResponseContextListener
+        arguments:
+            - '@contao.routing.scope_matcher'
+            - '@contao.routing.response_context_accessor'
+        tags:
+            # The priority must be lower than 'contao.listener.preview_toolbar' but higher than 'web_profiler.debug_toolbar'
+            - { name: kernel.event_listener, priority: -100 }
```

Comment:
Currently the response context is finalized in `PageRegular::getResponse`. However, this is too early if any `kernel.response` listeners also rely on response context functionality (including CSP). For example, our `PreviewToolbarListener` needs to be able to retrieve nonces as it injects `<script>` and `<style>` tags.

Thus the response context needs to be finalized _after_ the `PreviewToolbarListener` but _before_ Symfony's debug toolbar listener, as this one will generated its own nonces and merge them with existing CSP response headers.

Reply:
Changed in 0ceea3f50d